.github/workflows/build-full-application.yaml

```diff
@@ -0,0 +1,99 @@
+name: Build Full Application workflow
+
+on:
+  workflow_call:
+    inputs:
+      environment:
+        description: "The name of the environment to build into"
+        required: true
+        type: string
+      workspace:
+        description: "The name of the workspace to build the application into"
+        required: true
+        type: string
+      services:
+        description: "A list of the services to build"
+        required: false
+        default: ""
+        type: string
+      tag:
+        description: "The tag to build from"
+        required: false
+        default: ""
+        type: string
+      multi_domain:
+        description: "Domain to use for github runner role"
+        required: false
+        default: ""
+        type: string
+      workflow_timeout:
+        description: "Timeout duration in minutes"
+        required: false
+        default: 5
+        type: number
+      target_repo:
+        description: "The name of the repo"
+        required: true
+        type: string
+      target_repo_commit_hash:
+        description: "The name of the target repo branch"
+        required: true
+        type: string
+
+jobs:
+  deploy-infrastructure-apply:
+    name: "Apply Deploy Infrastructure"
+    strategy:
+      matrix:
+        domain: ['uec-dos-service-management','uec-cm']
+    uses: ./.github/workflows/deploy-infrastructure.yaml
+    with:
+      environment: ${{ inputs.environment }}
+      workspace: ${{ inputs.workspace }}
+      domain: ${{ matrix.domain }}
+      multi_domain: ${{ inputs.multi_domain }}
+      stacks: "['application']"
+      action: apply
+    secrets: inherit
+
+  deploy-lambda-code:
+    name: "Deploy Application "
+    needs:
+      [
+        deploy-infrastructure-apply,
+      ]
+    strategy:
+      matrix:
+        domain: ['uec-dos-service-management']
+    uses: ./.github/workflows/deploy-lambda-code.yaml
+    with:
+      environment: ${{ inputs.environment }}
+      workspace: ${{ inputs.workspace }}
+      custom_artefact_location: ${{ matrix.domain == inputs.target_repo && inputs.workspace || 'default' }}
+      domain: ${{ matrix.domain }}
+      multi_domain: ${{ inputs.multi_domain }}
+      services: 'all'
```

Comment:
Again Pira I think we discussed this yesterday. If this job is run once per domain listed in the domain matrix then identifying anything other than services: all would presumably only work for one of the domains listed and fail for the others. Eg if you specify healthcare-services-data-manager then it would work for service management but fail for eg service-search if we plugged that in.
Also if we are deploying the full application then why would we NOT deploy all services in a domaini. Would we want to cherry pick - genuine question ?
If services has to be all then we don't need to pass it do we ? We can just handle that in the deploy-lambda-code yaml ?

Reply:
The service being provided as an input to the deploy-lambda-code.yaml pre-dates the integration pipeline work, The intention is for the integration pipeline to always run with 'all', but the underlying deploy-lambda-code.yaml action will require a service input that may be different when its used in other circumstances.

We could change it so that we don't provide a service, it defaults to all, but I think we thought 'all' was more explicit / readable.

Reply:
Tim to resolve